extensions/wikia/Editcount/SpecialEditcount_body.php

```diff
@@ -198,27 +198,19 @@ function editsByNsAll( $uid ) {
 
 		if ( empty($nscount) ) {
 			$nscount = array();
-			if ( !empty( $wgStatsDBEnabled ) ) {
-				$dbs = wfGetDB(DB_SLAVE, array(), $wgStatsDB);
-				$res = $dbs->select(
-					array( 'events' ),
-					array( 'page_ns as namespace', 'count(page_ns) as count' ),
-					array(
-						'user_id' => $uid,
-						' ( event_type = 1 ) or ( event_type = 2 ) '
-					),
-					__METHOD__,
-					array (
-						'GROUP BY' => 'page_ns',
-						'ORDER BY' => 'null'
-					)
-				);
+			$dbs = wfGetDB(DB_SLAVE, array(), $wgDWStatsDB);
+			$res = $dbs->select(
+				array( 'rollup_wiki_namespace_user_events' ),
+				array( 'namespace_id as namespace', 'sum(creates + edits) as count' ),
+				array( 'user_id' => $uid, 'period_id' => DataMartService::PERIOD_ID_DAILY ),
+				__METHOD__,
+				array ( 'GROUP BY' => 'namespace_id' )
+			);
 
-				while( $row = $dbs->fetchObject( $res ) ) {
-					$nscount[$row->namespace] = $row->count;
-				}
-				$dbs->freeResult( $res );
+			while( $row = $dbs->fetchObject( $res ) ) {
+				$nscount[$row->namespace] = $row->count;
 			}
+			$dbs->freeResult( $res );
```

Comment:
Shouldn't be needed

Reply:
Why is there some mechanism that does that or we just don't care because PHP restarts the whole process for every request?

Reply:
According to method docs most of the time this method does not need to be called explicitly as standard GC is enough:

```
	/**
	 * Free a result object returned by query() or select(). It's usually not
	 * necessary to call this, just use unset() or let the variable holding
	 * the result object go out of scope.
	 *
	 * @param $res Mixed: A SQL result
	 */
	function freeResult( $res ) {
	}
```